base/compiler/typeutils.jl

```diff
@@ -56,6 +58,38 @@ function isknownlength(t::DataType)
     return isdefined(va, :N) && va.N isa Int
 end
 
+# Compute the minimum number of initialized fields for a particular datatype
+# (therefore also a lower bound on the number of fields)
+function datatype_min_ninitialized(t::DataType)
+    isabstracttype(t) && return 0
+    if t.name === NamedTuple_typename
+        names, types = t.parameters[1], t.parameters[2]
```

Comment:
We could just call `fieldcount` in this case.

Reply:
Hm, I guess not, since we want 0 instead of an error for abstract types...

Reply:
yes, that is why it is reimplemented here